assert/assertion_format.go

```diff
@@ -557,6 +562,14 @@ func WithinDurationf(t TestingT, expected time.Time, actual time.Time, delta tim
 	return WithinDuration(t, expected, actual, delta, append([]interface{}{msg}, args...)...)
 }
 
+// YAMLEqf asserts that two YAML strings are equivalent.
+func YAMLEqf(t TestingT, expected string, actual string, msg string, args ...interface{}) bool {
```

Comment:
@georgelesica-wf , @ernesto-jimenez : May I ask what the logic behind having this function in the package is? 

It seems a bit too close to a specific use-case rather than general testing business logic, in my opinion.

Reply:
Testing YAML and JSON equality is particularly difficult and that function and it's JSON counterpart makes it a lot easier and test output a lot easier to understand when two don't match. I don't think it is particularly business logic. Regardless, this function probably needs to stay if it's been added otherwise the change would be breaking?

Reply:
I understand the arguments/concerns about difficulty of testing JSON & YAML but my question arises from the fact that there are always new standards becoming the _de facto_ of the day :) 

I guess the argument I was trying to make is should the `testify` package also include standards-based formats or should be kept as a simple, low-level framework. I guess that's something for the steering group to decide :)

This won't be a breaking change by the way as one can only use the these functions in their code if they were `go generate`d. Since this code was never previously generated, it wouldn't have been used.

Reply:
Agreed that it's on the line, but in terms of philosophy I don't have strong feelings. Whatever the community finds helpful is probably fine as long as we don't go crazy.